pennylane/decomposition/symbolic_decomposition.py

```diff
@@ -16,238 +16,166 @@
 
 from __future__ import annotations
 
-import functools
+import numpy as np
 
 import pennylane as qml
 
-from .controlled_decomposition import base_to_custom_ctrl_op
 from .decomposition_rule import DecompositionRule, register_resources
 from .resources import adjoint_resource_rep, pow_resource_rep, resource_rep
+from .utils import DecompositionNotApplicable
 
 
-class AdjointDecomp(DecompositionRule):  # pylint: disable=too-few-public-methods
-    """The adjoint version of a decomposition rule."""
+def make_adjoint_decomp(base_decomposition: DecompositionRule):
```

Comment:
This is the same as before. What's previously a subclass is now a method. There was no reason why it had to be a subclass.